modules/jaggery-apps/app-workflow-admin/site/themes/default/templates/xacmlpolicy-task/js/xacmlpolicy.js

```diff
@@ -0,0 +1,432 @@
+            savePolicyPartial();
+            //alert("Policy is valid.", "alert-success");
+        } else {
+            alert("Policy is not valid.", "alert-danger");
+        }
+
+    } else {
+        alert("Could not complete validation.", "alert-danger");
+    }
+}
+
+
+function showMessageAfterValidation(response) {
+    var response = JSON.parse(response);
+    if (response.success) {
+        response = response.response;
+        if (response.isValid) {
+            alert("Policy is valid.", "alert-success");
+        } else {
+            alert("Policy is not valid.", "alert-danger");
+        }
+
+    } else {
+        alert("Could not complete validation.", "alert-danger");
+    }
+}
+
+
+function displayValidationRequestException() {
+    alert('Error occured while validating the policy', 'error');
+}
+
+
+function savePolicyPartial() {
+
+    var policyPartial = $('#policy-content').val();
+    var policyPartialName = $('#policy-name').val();
+    var policyPartialDesc = $('#policy-desc').val();
+
+    var provider = "";
+    var isSharedPartial = true;
+    if (editedpolicyPartialId == 0) { //add
+
+        $.ajax({
+            url: '/publisher/api/entitlement/policy/partial/save',
+            type: 'POST',
+            contentType: 'application/x-www-form-urlencoded',
+            async: false,
+            data: {
+                "policyPartialName": policyPartialName,
+                "policyPartial": policyPartial,
+                "isSharedPartial": isSharedPartial,
+                "policyPartialDesc": policyPartialDesc
+            },
+            success: function (data) {
+                var returnedId = JSON.parse(data).response.id;
+                editedpolicyPartialId = returnedId;
+                policyPartialsArray.push({
+                    id: returnedId,
+                    policyPartialName: policyPartialName,
+                    policyPartial: policyPartial,
+                    isShared: isSharedPartial,
+                    author: provider,
+                    description: policyPartialDesc
+                });
+                updatePolicyPartial()
+                alert("Policy Saved Successfully");
+                $('#policy-name').prop("readonly", true);
+            },
+            error: function () {
+            }
+        });
+
+    } else { // update
+        var policyPartialObj;
+
+        $.each(policyPartialsArray, function (index, obj) {
+            if (obj != null && obj.id == editedpolicyPartialId) {
+                policyPartialObj = obj;
+                return false; // break
+            }
+
+        });
+
+        $.ajax({
+            async: false,
+            url: '/publisher/api/entitlement/get/apps/associated/to/xacml/policy/id/' + editedpolicyPartialId,
+            type: 'GET',
+            contentType: 'application/json',
+            dataType: 'json',
+            success: function (response) {
+
+                console.info(JSON.stringify(response));
```

Comment:
This code was implemented during the parallel development with UI developer. This was fixed with a subsequent commit